src/editor/components/ErrorBoundary/ErrorBoundary.tsx

```diff
@@ -0,0 +1,48 @@
+import React, {ErrorInfo, PropsWithChildren} from 'react';
+
+import {BlockDecorationProps} from '../../../models';
+import {block} from '../../../utils';
+import {getBlockId} from '../../utils';
+
+import i18n from './i18n';
+
+import './ErrorBoundary.scss';
+
+const b = block('error-boundary');
+
+interface ErrorBoundaryProps extends PropsWithChildren, Partial<BlockDecorationProps> {}
+interface ErrorBoundaryState {
+    error?: string;
+}
+
+export class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
```

Comment:
Why are you using class instead of fc-component?

Reply:
React Error Boundary still doesn't have functional component realisation
https://ru.react.js.org/docs/error-boundaries.html